packages/open_aea/protocols/signing/tests/test_signing.py

```diff
@@ -48,191 +49,98 @@
-
-    def test_signed_transaction(self):
-        """Test for an error for a signed transaction."""
-        tx_msg = SigningMessage(
-            performative=SigningMessage.Performative.SIGNED_TRANSACTION,
-            message_id=2,
-            target=1,
-            signed_transaction=SignedTransaction(self.ledger_id, {"sig": "signature"}),
-        )
-        assert tx_msg._is_consistent()
-        encoded_tx_msg = tx_msg.encode()
-        decoded_tx_msg = tx_msg.serializer.decode(encoded_tx_msg)
-        assert tx_msg == decoded_tx_msg
-
-    def test_signed_message(self):
-        """Test for an error for a signed message."""
-        tx_msg = SigningMessage(
-            performative=SigningMessage.Performative.SIGNED_MESSAGE,
-            message_id=2,
-            target=1,
-            signed_message=SignedMessage(self.ledger_id, "message"),
-        )
-        assert tx_msg._is_consistent()
-        encoded_tx_msg = tx_msg.encode()
-        decoded_tx_msg = tx_msg.serializer.decode(encoded_tx_msg)
-        assert tx_msg == decoded_tx_msg
-
-    def test_error_message(self):
-        """Test for an error for an error message."""
-        tx_msg = SigningMessage(
-            performative=SigningMessage.Performative.ERROR,
-            message_id=2,
-            target=1,
-            error_code=SigningMessage.ErrorCode.UNSUCCESSFUL_MESSAGE_SIGNING,
-        )
-        assert tx_msg._is_consistent()
-        encoded_tx_msg = tx_msg.encode()
-        decoded_tx_msg = tx_msg.serializer.decode(encoded_tx_msg)
-        assert tx_msg == decoded_tx_msg
-        assert str(tx_msg.performative) == "error"
-        assert len(tx_msg.valid_performatives) == 5
-
-
-def test_consistency_check_negative():
-    """Test the consistency check, negative case."""
-    tx_msg = SigningMessage(
-        performative=SigningMessage.Performative.SIGN_TRANSACTION,
-    )
-    assert not tx_msg._is_consistent()
-
-
-def test_serialization_negative():
-    """Test serialization when performative is not recognized."""
-    tx_msg = SigningMessage(
-        performative=SigningMessage.Performative.ERROR,
-        message_id=2,
-        target=1,
-        error_code=SigningMessage.ErrorCode.UNSUCCESSFUL_MESSAGE_SIGNING,
-    )
-
-    with patch.object(SigningMessage.Performative, "__eq__", return_value=False):
-        with pytest.raises(
-            ValueError, match=f"Performative not valid: {tx_msg.performative}"
-        ):
-            tx_msg.serializer.encode(tx_msg)
-
-    encoded_tx_bytes = tx_msg.serializer.encode(tx_msg)
-    with patch.object(SigningMessage.Performative, "__eq__", return_value=False):
-        with pytest.raises(
-            ValueError, match=f"Performative not valid: {tx_msg.performative}"
-        ):
-            tx_msg.serializer.decode(encoded_tx_bytes)
+class TestMessages(BaseProtocolMessagesTestCase):
+    """Base class to test message construction for the protocol."""
 
+    __test__ = True
```

Comment:
with the disclaimed that I never used this `__test__` feature, but I believe you don't need to provide this overwrite. Please confirm. If so would actually be suboptimal as the default would now be that such tests would not be run, hence forgetting to add them or accidentally removing such an overwrite would disable tests in a way that might easily go unnoticed.

Reply:
I also hope we don't need this. If we do need it then the metaclass should check it's been activated on the concrete class.

Reply:
removed. it was done for extra test of the test case. mostly like experiment.
now i see it brings more questions than actual help. so removed this